packages/neon-api/src/transaction/validator.ts

```diff
@@ -0,0 +1,229 @@
+          fixed: false,
+          prev: validUntilBlock,
+          suggestion: vub_suggestion
+        }
+      };
+    }
+    return {
+      valid: true
+    };
+  };
+
+  // validateIntents = async (): Promise<ValidationResult> => {
+  //   return {
+  //     result: true,
+  //     fixed: false
+  //   };
+  // };
+
+  /**
+   * validate systemFee
+   * @param autoFix will automatically fix transaction if specified as true
+   */
+  validateSystemFee = async (autoFix = false): Promise<ValidationResult> => {
+    const { script, systemFee } = this.transaction;
+    const { gas_consumed } = await this.rpcClient.invokeScript(script);
+    const minimumSystemFee = new u.Fixed8(parseFloat(gas_consumed)).ceil();
+    if (autoFix && !minimumSystemFee.equals(systemFee)) {
+      this.transaction.systemFee = minimumSystemFee;
+      return {
+        valid: true,
+        systemFee: {
+          fixed: true,
+          prev: systemFee,
+          suggestion: minimumSystemFee
+        }
+      };
+    } else if (minimumSystemFee.isGreaterThan(systemFee)) {
+      return {
+        valid: false,
+        systemFee: {
+          fixed: false,
+          prev: systemFee,
+          suggestion: minimumSystemFee
+        }
+      };
+    }
+    return {
+      valid: true
+    };
+  };
+
+  /**
+   * Validate NetworkFee
+   * @param autoFix will automatically fix transaction if specified as true
+   */
+  validateNetworkFee = async (autoFix = false): Promise<ValidationResult> => {
+    const { networkFee } = this.transaction;
+    const minimumNetworkFee = getNetworkFee(this.transaction);
+    if (autoFix && !minimumNetworkFee.equals(networkFee)) {
+      this.transaction.networkFee = minimumNetworkFee;
+      return {
+        valid: true,
+        networkFee: {
+          fixed: true,
+          prev: networkFee,
+          suggestion: minimumNetworkFee
+        }
+      };
+    } else if (minimumNetworkFee.isGreaterThan(networkFee)) {
+      return {
+        valid: false,
+        networkFee: {
+          fixed: false,
+          prev: networkFee,
+          suggestion: minimumNetworkFee
+        }
+      };
+    }
+    return {
+      valid: true
+    };
+  };
+
+  /**
+   * Validate Signatures
+   */
+  validateSigning = async (): Promise<ValidationResult> => {
+    const scriptHashes: string[] = getScriptHashesFromTxWitnesses(
+      this.transaction
+    );
+    const signers = this.transaction
+      .getScriptHashesForVerifying()
+      .map(hash => u.reverseHex(hash));
+
+    let notSignedHash = "";
+    if (
+      signers.every(signer => {
+        if (scriptHashes.indexOf(signer) < 0) {
+          notSignedHash = signer;
+          return false;
+        }
+        return true;
+      })
+    ) {
+      return {
+        valid: true
+      };
+    }
+
+    return {
+      valid: false,
+      witnesses: {
+        notSignedHash
+      }
+    };
+  };
+
+  async validate(
+    attrs: ValidationAttributes,
+    autoFix = false
```

Comment:
can we also have `autoFix` be `ValidationAttributes` so we can select properties to autofix?

Reply:
Emm.. then the flag thing should be changed, right?
I believe user can invoke twice, one with autoFix, one without to achieve this scope.

Reply:
> Emm.. then the flag thing should be changed, right?

What do you mean? 

> I believe user can invoke twice, one with autoFix, one without to achieve this scope

Yes they could. I think the autoFix change is a small natural improvement using the same set of flags.

Reply:
Oh, I see it. Will do.

Reply:
`autoFix: ValidationAttributes | boolean`
Give user the freedom to assign one boolean, too.

Reply:
you can create the convenient flags `ValidationAttributes.ALL` and  `ValidationAttributes.NONE`